### ISSUE TITLE
Is timeoutFailure used?

### ISSUE THREAD

**Issue by joemasilotti** [hotwired/turbo#237]
I only see [two mentions of the error](https://github.com/hotwired/turbo/search?q=timeoutFailure): the definition and the mapping for adapters.

Is this error used anywhere? If so, where?

**Comment by sstephenson** [comment]
Good catch! Looks like we didn't implement request timeout when we moved from XHR to Fetch.

**Comment by sstephenson** [comment]
https://github.com/turbolinks/turbolinks/blob/80cd172ed5b3a35eda85c58e428de9f1f1d06fc8/src/http_request.ts#L127

**Comment by joemasilotti** [comment]
Apparently `fetch()` doesn't have built in time-out support! Leaving this comment on how to roll-your-own for future reference.

https://github.com/github/fetch/issues/175#issuecomment-125779262

```javascript
function timeout(ms, promise) {
  return new Promise(function(resolve, reject) {
    setTimeout(function() {
      reject(new Error("timeout"))
    }, ms)
    promise.then(resolve, reject)
  })
}

timeout(1000, fetch('/hello')).then(function(response) {
  // process response
}).catch(function(error) {
  // might be a timeout error
})
```
